dataactbroker/helpers/fabs_derivations_helper.py

```diff
@@ -362,10 +362,12 @@ def derive_ppop_location_data(sess, submission_id):
     # Deriving city info for transactions with zips
     query = """
         UPDATE tmp_fabs_{submission_id}
-        SET place_of_performance_city = city_name
+        SET place_of_performance_city = preferred_city_name
         FROM zip_city
         WHERE place_of_performance_zip5 IS NOT NULL
-            AND zip_city.zip_code = place_of_performance_zip5;
+            AND zip_city.zip_code = place_of_performance_zip5
+            AND place_of_perfor_state_code IS NOT NULL
```

Comment:
This is extraneous. If it's NULL it'll just not match on anything

Reply:
Should the `place_of_performance_zip5 IS NOT NULL`s be removed as well?

Reply:
Those can be left. If they're not null it's most likely that the state will also not be null. One check can narrow the list, the second check will end up being extraneous.